Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs

```diff
@@ -126,6 +129,8 @@ select personId estimateId req@DSelectReq {..} = do
   now <- getCurrentTime
   estimate <- QEstimate.findById estimateId >>= fromMaybeM (EstimateDoesNotExist estimateId.getId)
   let searchRequestId = estimate.requestId
+  person <- QP.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
```

Comment:
Check for `isValueAddNP` here, so that this lookup is not wasted in case of off-us transaction.

If off-us then you can pass `Nothing` here itself